QubitServer/src/main/java/org/labrad/qubits/Device.scala

```diff
@@ -0,0 +1,96 @@
+package org.labrad.qubits
+
+import com.google.common.collect.Lists
+import com.google.common.collect.Maps
+import java.util.List
+import java.util.Map
+import org.labrad.qubits.channels.Channel
+import scala.collection.JavaConverters._
+
+/**
+ * Model of a device, essentially a collection of named channels of various types.
+ * Each channel connects various physical devices (FastBias, Preamp, GHzDac) to some
+ * controllable parameter on the device (flux bias, microwave, etc.).
+ * 
+ * @author maffoo
+ */
+class Device(name: String) {
+
+  private val channels: List[Channel] = Lists.newArrayList()
+  private val channelsByName: Map[String, Channel] = Maps.newHashMap()
+
+  def getName(): String = {
+    name
+  }
```

Comment:
Note that in the next change I have queued up, these `getName` methods just go away entirely, because if you have a public member in scala, you get a generated 'accessor' method with the same name. Then you can just say `device.name` instead of `device.getName()`. I think some of the weirdness here might just be due to the fact that in trying to make these changes incremental (for easier reviewing, I hope), we end up straddling the java/scala worlds in ways that are a little foreign to both languages (see also the collections conversions).